Statement.php

```diff
@@ -147,17 +156,21 @@ public function errorInfo() {
 
     /**
      * {@inheritdoc}
+     *
+     * @SuppressWarnings("PHPMD.StaticAccess")
      */
     public function execute($params = null) {
-        $rawRequest = $this->mysqlToRequest->transform($this->query, $this->params);
-        $request    = $this->authStrategy->transformRequest($rawRequest);
-        $restClient = $this->restClientFactory->createOne($request->getCurlOptions());
+        $transformedQuery = SqlQuery::quoteUrl(SqlQuery::setParams($this->query, $params !== null ? $params : $this->params));
+        $rawRequest       = $this->mysqlToRequest->transform($transformedQuery);
+        $request          = $this->authStrategy->transformRequest($rawRequest);
+
+        $response = $this->restClient->send($request);
 
-        $method     = strtolower($request->getMethod());
-        $response   = $method === HttpMethods::GET || $method === HttpMethods::DELETE ? $restClient->$method($request->getUrlAndQuery()) : $restClient->$method($request->getUrlAndQuery(), $request->getPayload());
-        $statusCode = $response->getStatusCode();
+        $this->result = Result::create($transformedQuery, $this->formatter->decode($response->getContent()));
+        $this->id     = !empty($this->result['id']) ? $this->result['id'] : null;
+        krsort($this->result);
```

Comment:
was soll das hier? und egal was es macht, gehört das nich in result.create?

dann zeile da drüber: soweit ich das sehe, is doch diese id entweder null oder die id und du mach da folgendes: du checks, is es null, dann weise null zu oder nich, dann this.result.id. Soweit ich das sehe, würde hier reichen:

``php
$this->id = $this->result("id");
```